docs/ql-documentation/learn-ql/javascript/flow-labels.rst

```diff
@@ -0,0 +1,369 @@
+Advanced data-flow analysis using flow labels
+=============================================
+
+You can use basic inter-procedural data-flow analysis and taint tracking as described in
+:doc:`Analyzing data flow in JavaScript/TypeScript <dataflow>` to check whether there is a path in
+the data-flow graph from some source node to a sink node that does not pass through any sanitizer
+nodes. Another way of thinking about this is that it statically models the flow of data through the
+program, and associates a flag with every data value telling us whether it might have come from a
+source node.
+
+Limitations of basic data-flow analysis
+---------------------------------------
+
+In many applications we are interested in tracking more than just the reachability information provided by inter-procedural data flow analysis.
+
+For example, when tracking object values that originate from untrusted input, we might want to
```

Comment:
Add a link to the prototype pollution query as an example.

Reply:
I've added a link for this query and the tainted-path one to the end of the document, since I felt it would break the flow to add them here.